app/ts/store/balance.ts

```diff
@@ -0,0 +1,48 @@
+import { signal } from '@preact/signals'
+import { ethers } from 'ethers'
+import { assertsExternalProvider } from '../library/utilities.js'
+import { accountStore } from './account.js'
+
+type BalanceStore =
+	| {
+			state: 'inactive'
+			checkEthBalance: () => Promise<void>
+	  }
+	| {
+			state: 'pending'
+			reset: () => void
+	  }
+	| {
+			state: 'rejected'
+			error: Error
+			reset: () => void
+	  }
+	| {
+			state: 'resolved'
+			balance: string
+			lastChecked: Date
+			reset: () => void
+	  }
+
+async function checkEthBalance() {
+	try {
+		assertsExternalProvider(window.ethereum)
+		const provider = new ethers.providers.Web3Provider(window.ethereum)
+		if (accountStore.value.status !== 'connected') throw new Error('Not connected!')
+		balanceStore.value = { state: 'pending', reset }
+		const balanceBigNumber = await provider.getBalance(accountStore.value.address)
+		const balance = ethers.utils.formatEther(balanceBigNumber)
+		balanceStore.value = { state: 'resolved', balance, lastChecked: new Date(), reset }
```

Comment:
you should store ether numbers as a big int and only when doing rendering, show it as a string